app/controllers/ruleset_controller.rb

```diff
@@ -30,38 +30,27 @@ def create
       return render @ruleset_name
     end
 
-    # Make the API calls in a begin/rescue block so that
-    # we don't crash out if the API or CLI calls fail.
-    begin
-      if @ruleset_name == 'security_ruleset' # Call 42Crunch service
-        if ENV['BYPASS_API'] == 'true'
-          crunch_results = File.read('fakecrunchresults.json')
-        else
-          crunch_results = Linters::CrunchApi.new.lint_to_json(ruleset_params[:oas_file].tempfile)
-        end
-        crunch_json = JSON.parse(crunch_results)
-        @score = crunch_json['score']
-        @openapi = crunch_json['openapiState']
-        @counter = crunch_json['issueCounter']
-        @criticality = crunch_json['criticality']
-        @issues = Array.new
-        crunch_json['data']['issues'].each do |issue|
-          lines = Array.new
-          issue[1]['issues'].each do | occurrence|
-            lines.push(occurrence['pointer'])
-          end
-          @issues.push([issue[1]['criticality'],issue[1]['description'], lines])
+    if @ruleset_name == 'security_ruleset' # Call 42Crunch service
+      if ENV['BYPASS_API'] == 'true'
+        crunch_results = File.read('fakecrunchresults.json')
+      else
+        crunch_results = Linters::CrunchApi.new(file: ruleset_params[:oas_file]).lint_to_json
+      end
+      crunch_json = JSON.parse(crunch_results)
+      @score = crunch_json['score']
+      @openapi = crunch_json['openapiState']
+      @counter = crunch_json['issueCounter']
+      @criticality = crunch_json['criticality']
+      @issues = Array.new
+      crunch_json['data']['issues'].each do |issue|
+        lines = Array.new
+        issue[1]['issues'].each do | occurrence|
+          lines.push(occurrence['pointer'])
         end
-      elsif @ruleset_name == 'government_ruleset' # Call Spectral service
-        @spectral_results = JSON.parse(Linters::Spectral.new(file: ruleset_params[:oas_file]).lint_to_json)
+        @issues.push([issue[1]['criticality'],issue[1]['description'], lines])
       end
-    rescue StandardError => e
-      Rails.logger.debug e.message
-      # TODO: We don't want to reveal the full 42Crunch or Spectral error message
-      # to the user but we might for example give the user a reference number here
-      # so that we can find the full error in the logs.
-      #flash[:error] = "Unable to run linter on the uploaded file"
-      flash[:error] = e.message
+    elsif @ruleset_name == 'government_ruleset' # Call Spectral service
+      @spectral_results = JSON.parse(Linters::Spectral.new(file: ruleset_params[:oas_file]).lint_to_json)
```

Comment:
If like I said before we want to use hashes, then this will probably be better of being returned as a hash  from `Linter::Spectral` rather than json.

Reply:
Agreed, but I think we need to think a bit about the data commonalities between 42crunch and spectral and see if we can have a common structure for the data. So this is just about not changing too much at once really.